essentials-framework/Essentials Core/PepperDashEssentialsBase/Display/DisplayBase.cs

```diff
@@ -179,32 +177,32 @@ public virtual FeedbackCollection<Feedback> Feedbacks
                 }
             };
 
-            displayDevice.PowerIsOnFeedback.LinkComplementInputSig(trilist.BooleanInput[joinMap.PowerOff]);
+            displayDevice.PowerIsOnFeedback.LinkComplementInputSig(trilist.BooleanInput[joinMap.PowerOff.JoinNumber]);
 
             // PowerOn
-            trilist.SetSigTrueAction(joinMap.PowerOn, () =>
+            trilist.SetSigTrueAction(joinMap.PowerOn.JoinNumber, () =>
             {
                 inputNumber = 0;
                 inputNumberFeedback.FireUpdate();
                 displayDevice.PowerOn();
             });
 
 
-            displayDevice.PowerIsOnFeedback.LinkInputSig(trilist.BooleanInput[joinMap.PowerOn]);
+            displayDevice.PowerIsOnFeedback.LinkInputSig(trilist.BooleanInput[joinMap.PowerOn.JoinNumber]);
 
-            var count = 1;
+            var count = 0;
             foreach (var input in displayDevice.InputPorts)
```

Comment:
Here we likely need to perform some check that we don't exceed the span defined in the join map.  Might be better to use a for loop than foreach and first check what the displayDevice.InputPorts.Count value is and see if it's greater than the joinMap.InputSelectNumber.JoinSpan value

Reply:
Also, the big clue here is that if we need a count variable outside the foreach, we should probably use a for loop instead.

Reply:
Im not sure there's an easy way to do a for loop here.  It's a dictionary.

I think the current implementation is fine as long as I add a check for the count as a function of the joinSpan.

`var count = 0;
foreach (var input in displayDevice.InputPorts)
{
    if (count < joinMap.InputSelectOffset.JoinSpan)
    {
        inputKeys.Add(input.Key);
        var tempKey = inputKeys.ElementAt(count - 1);
        trilist.SetSigTrueAction((ushort)(joinMap.InputSelectOffset.JoinNumber + count), () => displayDevice.ExecuteSwitch(displayDevice.InputPorts[tempKey].Selector));
        Debug.Console(2, displayDevice, "Setting Input Select Action on Digital Join {0} to Input: {1}", joinMap.InputSelectOffset.JoinNumber + count, displayDevice.InputPorts[tempKey].Key.ToString());
        trilist.StringInput[(ushort)(joinMap.InputNamesOffset.JoinNumber + count)].StringValue = input.Key.ToString();
        count++;
    }
}`

Reply:
Gross Formatting Just check my changes when I push it.